### PR TITLE
fix: typing

### PR DESCRIPTION
**Pull request by marcosschroh**

### CODE REVIEW

File: kstreams/engine.py [hunk 1]
```diff
@@ -59,18 +64,19 @@ async def send(
                 value, headers=headers, value_serializer_kwargs=value_serializer_kwargs
             )
 
+        kafka_headers = None
         if headers is not None:
-            headers = encode_headers(headers)
+            kafka_headers = encode_headers(headers)
```
Comment (Woile) [inline kstreams/engine.py:69]: I would remove direct references to kafka if possible in the code outside `backends.Kafka`

File: kstreams/exceptions.py [hunk 2]
```diff
@@ -13,3 +13,8 @@ def __str__(self) -> str:
         )
 
         return msg
+
+
+class EngineNotStartedException(Exception):
+    def __str__(self) -> str:
+        return "Engine has not been started. Try with `await engine.start()`"
```
Comment (Woile) [inline kstreams/exceptions.py:20]: nice!

File: kstreams/custom_types.py [hunk 3]
```diff
@@ -1,5 +1,4 @@
-from typing import Any, Callable, Dict, Sequence, Tuple, TypeVar
+from typing import Dict, Sequence, Tuple
```
Comment (Woile) [inline kstreams/custom_types.py:1]: Maybe this file should be called `types`. From the user perspective, importing `custom_types` may be confusing. I would expect to import the types from kstreams. See [starlette types](https://github.com/encode/starlette/blob/master/starlette/types.py)

File: kstreams/engine.py (hunk 1 above)
Comment (marcosschroh) [reply kstreams/engine.py:69]: yes, it makes sense

File: kstreams/custom_types.py (hunk 3 above)
Comment (marcosschroh) [reply kstreams/custom_types.py:1]: Yes, it is confusing. If the `rename` does not introduce too many changes I will do it

**Review by Woile — APPROVED** [review]

**Pull request merged**